main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1591,6 +1592,28 @@ private void updateChirpWolfBox() {
                 });
             }
         }
+
+        private void updateALCBox() {
+            final boolean isEnabled = cache.getType() == CacheType.ADVLAB && StringUtils.isNotEmpty(cache.getUrl());
+            final Intent alc = ProcessUtils.getLaunchIntent(getString(R.string.package_alc));
+            final LinearLayout boxCW = view.findViewById(R.id.alc_box);
+            boxCW.setVisibility(isEnabled ? View.VISIBLE : View.GONE);
+            final TextView msgCW = view.findViewById(R.id.alc_text);
+            msgCW.setText(null != alc ? R.string.cache_alc_start : R.string.cache_alc_install);
```

Comment:
```suggestion
            msgCW.setText(alc != null ? R.string.cache_alc_start : R.string.cache_alc_install);
```
Personally, I like to have `null` on the right-hand side because "value is not null" reads more natural than "null is not value".

Reply:
Me too. I have copy / pasted it from Chrip and I did that for consistency. Now if I change it for ALC, do you also want me to change it for Chirp?

Reply:
I propose to fix that separately, as it's unrelated to this PR, and there are probably many more place where we have "null" at the left hand side. We should tackle them all in one separate PR (except those "introduced" with this PR, those should be fixed here).

Reply:
since I had to commit it for the ALC related part, I fixed Chirp too